cache.go

```diff
@@ -201,9 +201,7 @@ func (c *Cache[K, V]) getWithOpts(key K, opts ...Option[K, V]) *Item[K, V] {
 
 	applyOptions(&getOpts, opts...)
 
-	c.items.mu.Lock()
 	elem := c.get(key, !getOpts.disableTouchOnHit)
-	c.items.mu.Unlock()
```

Comment:
One problem with this change is that the loader's `Load()` (line 214) will execute while the mutex is still locked, which might affect the performance of the whole cache (e.g., what if the loader performs some long-running/HTTP operations?)

Reply:
any idea how to address this? @DoubleDi 

cc @davseby

Reply:
Hi! You have a good point about the Load function. I am not sure there is a way to keep the get atomic and use the loader.

Maybe we can sidestep the problem and add a singleflight on the GetOrSet, GetAndDelete methods. 

Let me give you an example
```
func (c *Cache[K, V]) GetOrSet(key K, value V, opts ...Option[K, V]) (*Item[K, V], bool) {
    return c.sf.Do(key, func() {
            return c.getOrSet(key, value, opts...)
    })
}
```

That way even though the calls are still not atomic, for every key all the users will receive a determinate result.

Reply:
We should probably skip the loading step as in both cases we will be overriding the value, either by replacing or deleting it. Perhaps, we could introduce a bool parameter to skip the loading step.

Reply:
I assume this is not applicable to GetAndDelete. If we skip the load phase, we will not get any data from the cache, but we should return the data from Load

Reply:
let me try to takle the problems by merging the logic from get and delete, get and set in both methods

Reply:
Interesting point, however it begs the question, how do we treat the loader in case of the `GetOrSet`? Is the loader the extension of get, where if we retrieve the value via loader (and set it), we don't set a new one? Based on the current `Get` implementation, we should probably load it.

Perhaps, a new Option could be added that would suppress the loading? @swithek

Reply:
`WithLoader(nil)` could be used to suppress the active loader, though I'm not too sure whether I'd like this approach. I think the latest commit and my comment below outline how we should proceed.